### PR TITLE
Avoid copy action for root assemblies

### LINKED ISSUE #95429: Linker doesn't substitute feature switches when used with 'copy'
Our recommended way to get all warnings from a library is to use a sample app that references the library with the 'copy' action, as described in https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/prepare-libraries-for-trimming

Unfortunately, it doesn't seem like feature switches work as expected in this scenario. Specifically, the properties don't get replaced and the warnings don't get suppressed.

### PR DESCRIPTION
**Pull request by sbomer**
Changes `TrimmerRootAssembly` (and the underlying option `-a`) to root all members of the assembly, without setting the action to `copy`. This allows us to perform branch removal and attribute removal for the assembly.

Fixes https://github.com/dotnet/runtime/issues/95429

### CODE REVIEW

**Comment by lambdageek** [comment]
> The ApplyUpdateTests are running for mono wasm in Release mode, where DebuggerSupport was set to false by default. This illink change results in DebuggableAttribute being removed, causing test failures, so I set DebuggerSupport to true for these tests to preserve the attribute. Now I am getting failures which I don't know how to diagnose, and I'm not sure whether they're expected. For example, this assert is hitting:
> 
> https://github.com/dotnet/runtime/blob/89af7af9d9f477e7969297aacce10c17c1f29ca0/src/libraries/System.Runtime.Loader/tests/ApplyUpdateTest.cs#L226
> 
> @lambdageek Could you please advise? Would it be better to simply disable these tests in Release mode, or are these failures worth investigating?
> 
> edit: It seems those failures aren't reproing in CI - maybe I was doing something incorrectly locally.

@sbomer The EnC tests depend at runtime on the assemblies built from the projects in the `ApplyUpdate` subdirectory.  Those assemblies need to be compiled by Roslyn with optimizations off and with emitted debug information see https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime.Loader/tests/ApplyUpdate/Directory.Build.props

(specifically when a runtime that supports hot reload tries to apply an update to a particular assembly, that assembly must have `[assembly:System.Diagnostics.DebuggableAttribute(DebuggingModes.DisableOptimizations)]`).

In general the assemblies in the `ApplyUpdate` directory must not be modified by the trimmer at all - otherwise the hot reload deltas will not match the assembly contents and the runtime will do something nonsensical. (in real world usage the deltas are generated by VS; for the tests it's the `*.dll.1.dmeta` and `*.dll.1.dil` files that are generated during the test build ahead of time).  You can think of the hot reload deltas as some kind of log that says "add a new row N to table X with contents ABCD; modify row M of table Y with contents SPQR; etc" - if the trimmer disturbs the baseline assembly at all - things will go wrong.

File: src/libraries/System.Runtime.Loader/tests/System.Runtime.Loader.Tests.csproj [hunk 1]
```diff
@@ -2,6 +2,7 @@
   <PropertyGroup>
     <RootNamespace>System.Runtime.Loader.Tests</RootNamespace>
     <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
+    <DebuggerSupport Condition="'$(DebuggerSupport)' == '' and '$(TargetOS)' == 'browser'">true</DebuggerSupport>
```
Comment (MichalStrehovsky) [inline src/libraries/System.Runtime.Loader/tests/System.Runtime.Loader.Tests.csproj:5]: Is this still needed with the target below?

File: src/libraries/System.Runtime/tests/System.Runtime.Tests/System/Attributes.cs [hunk 2]
```diff
@@ -327,6 +328,8 @@ public static class GetCustomAttribute
     {
 
         [Fact]
+        [ActiveIssue("https://github.com/dotnet/linker/issues/2078", typeof(PlatformDetection), nameof(PlatformDetection.IsTrimmedWithILLink))
```
Comment (MichalStrehovsky) [inline src/libraries/System.Runtime/tests/System.Runtime.Tests/System/Attributes.cs:331]: Wondered why this wasn't hit with naot and found out illinker has this as a hardcoded list. Weird, but okay I guess. Would be nice to move this to the same plan as all the other attributes at least. I'm not convinced anyone will want to work on 2078 anytime soon.

File: src/tools/illink/src/linker/Linker.Steps/UnreachableBlocksOptimizer.cs [hunk 3]
```diff
@@ -1583,7 +1583,7 @@ public bool Analyze (in CalleePayload callee, Stack<MethodDefinition> callStack)
 								return false;
 
 							if (operand is int oint) {
-								if (oint == 1)
+								if (oint != 0)
```
Comment (MichalStrehovsky) [inline src/tools/illink/src/linker/Linker.Steps/UnreachableBlocksOptimizer.cs:1586]: Why was this needed? It's a good change, but surprised to see this in this PR.

**Review by MichalStrehovsky — APPROVED** [review]
Looks reasonable to me although I'm not an expert on this codebase.
